server/zio-http-server/src/main/scala/sttp/tapir/server/ziohttp/ZioHttpToResponseBody.scala

```diff
@@ -36,23 +36,26 @@ class ZioHttpToResponseBody extends ToResponseBody[ZioHttpResponseBody, ZioStrea
       case RawBodyType.ByteArrayBody   => (ZStream.fromChunk(Chunk.fromArray(r)), Some((r: Array[Byte]).length.toLong))
       case RawBodyType.ByteBufferBody  => (ZStream.fromChunk(Chunk.fromByteBuffer(r)), None)
       case RawBodyType.InputStreamBody => (ZStream.fromInputStream(r), None)
+      case RawBodyType.InputStreamRangeBody => 
+        r.range
+        .map(range => (ZStream.fromInputStream(r.inputStreamFromRangeStart()).take(range.contentLength), Some(range.contentLength)))
+        .getOrElse((ZStream.fromInputStream(r.inputStream()), None))
       case RawBodyType.FileBody =>
-        val tapirFile = r.asInstanceOf[FileRange]
-        val stream = tapirFile.range
+        val tapirFile = r
+        tapirFile.range
           .flatMap { r =>
             r.startAndEnd.map { s =>
               var count = 0L
-              ZStream
+              (ZStream
                 .fromPath(tapirFile.file.toPath)
                 .dropWhile(_ =>
                   if (count < s._1) { count += 1; true }
                   else false
                 )
-                .take(r.contentLength)
+                .take(r.contentLength), Some(r.contentLength))
             }
           }
-          .getOrElse(ZStream.fromPath(tapirFile.file.toPath))
-        (stream, Some(tapirFile.file.length))
```

Comment:
**question:** I believe this was incorrect, so I changed `file.length` to `r.contentLength`. This is the case where we handle ranges, so we shouldn't return total file size, right?

Reply:
yeah this seems correct; should be the same in other interpreters